src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlSequentialTextReader.cs

```diff
@@ -172,7 +172,7 @@ public override Task<int> ReadAsync(char[] buffer, int index, int count)
                         byte[] byteBuffer = PrepareByteBuffer(charsNeeded, out int byteBufferUsed);
 
                         // Permit a 0 byte read in order to advance the reader to the correct column
-                        if ((byteBufferUsed < byteBuffer.Length) || (byteBuffer.Length == 0))
+                        if ((byteBufferUsed <= byteBuffer.Length) || (byteBuffer.Length == 0))
```

Comment:
nit: remove redundant parenthesis while you're here

Reply:
Done.